### PR TITLE
bug fix: undefined method 'author' for nil:NilClass on new user profile

### LINKED ISSUE #9884: Error on visiting own profile by a new user(user without posts)
undefined method `author' for nil:NilClass error on visiting own profile by a new user
image

### PR DESCRIPTION
**Pull request by 17sushmita**
Fixes #9884

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by jywarren** [comment]
Sorry this is so annoying! I actually just merged https://github.com/publiclab/plots2/pull/9881 so if you'd like to try rebasing over it it might help!

File: app/views/tag/_tagging.html.erb [hunk 1]
```diff
@@ -48,7 +48,7 @@ $(function () {
 <% parent ||= nil %>
 
 <% if current_user && (parent != :profile || (current_user.id == user.id || logged_in_as(['admin']))) && !@preview  %>
-  <% if current_user.first_time_poster && !(current_user.username == @node.author.username || ( @node.coauthors && @node.coauthors.exists?(username: current_user.username)) || logged_in_as(['admin', 'moderator'])) %>
+  <% if current_user.first_time_poster && !(parent == :profile || current_user.username == @node.author.username || ( @node.coauthors && @node.coauthors.exists?(username: current_user.username)) || logged_in_as(['admin', 'moderator'])) %>
```
Comment (cesswairimu) [inline app/views/tag/_tagging.html.erb:51]: hi @17sushmita ,  maybe we could try use nil navigator `@node&.author&.username`  https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

Comment (17sushmita) [reply app/views/tag/_tagging.html.erb:51]: Hi @cesswairimu, Actually the `@node` even if present would be referring to the nodes written by the user. I think this tag moderation is relevant to only notes page and not the profile page.
If we also do tag moderation to the profile page, new user would not be able to add tags to his own profile. e.g-
image

And for nodes page, `@node` will never be nil, so the nil navigator will not be required.
Did I miss any of the case? What do you think?

Comment (cesswairimu) [reply app/views/tag/_tagging.html.erb:51]: Aha! gotcha, thanks for the clarification @17sushmita, I will try to start rebuild

Comment (cesswairimu) [reply app/views/tag/_tagging.html.erb:51]: Looks like we are all good here.. let me know if we are good to merge. Thanks

Comment (17sushmita) [reply app/views/tag/_tagging.html.erb:51]: yes @cesswairimu. It is ready to merge.

**Review by cesswairimu — APPROVED** [review]
:tada: :tada: Thanks @17sushmita

**Comment by jywarren** [comment]
Great work!!!

**Pull request merged**